### DISCUSSION TITLE
UPDI programming works on < 9600bps, fails on anything faster

### DISCUSSION THREAD

**Discussion by usr97629238** [SpenceKonde/megaTinyCore#1128] [Q&A] [answered]
Unfortunately I **don't have an oscilloscope**, so please advise what could be wrong here, as I don't have any ideas short of just shuffling bigger/smaller resistors (and they are already soldered) until something works.
Should I unsolder the TX LED?
Is the use of 1N5817WS instead of BAT54 problematic?
Also it only works with 3.3v (VCC/TX), on 5.0v it fails even at 1200bps.

The chip is attiny402 and there are no connections to it besides UPDI/VCC/GND.
The USB/UART chip is CH340G.
The megaTinyCore/prog.py is 2.6.10.
The connection is like this:
```
--------------------
                DTR|                                  __________________
                Rx |--------------┬------------------| UPDI---\/\/---------->
  Tx---/\/\/\---Tx |--┬----|<|----┘                  |        470
         1k        |  |  1N5817WS                    |
        	   |  |                              |
        	   |  └--LED---\/\/--┐	             |
                   |           470   |      ┌--------| Gnd
                Vcc|-----------------┴------┼--------| Vcc
                CTS|                        |        |
                Gnd|------------------------┘        |__________________
--------------------
```

**Comment by hmeijdam** [comment, marked as answer]
I suspect TX has struggles to pull the UPDI low enough.

You could try to swap the resistors. 1K to the led and 470 in series with TX. Or did you try that already while shuffling resistors?

**Comment by usr97629238** [comment]
Thanks, that did it, running fine at 921600bps with no delays.

So, if you have one of those:
![konwerter-usb-ttl-ch340-b-o-3_1648](https://github.com/user-attachments/assets/cccd3475-dec6-450a-84a6-fcb9d718d6e6)

wire them like:
```
--------------------
                DTR|                                  __________________
                Rx |--------------┬------------------| UPDI---\/\/---------->
  Tx---/\/\/\---Tx |--┬----|<|----┘                  |        1k
         470       |  |  1N5817WS                    |
        	   |  |                              |
        	   |  └--LED---\/\/--┐	             |
                   |           470   |      ┌--------| Gnd
                Vcc|-----------------┴------┼--------| Vcc
                CTS|                        |        |
                Gnd|------------------------┘        |__________________
--------------------
```

**Reply by hmeijdam** [reply]
Did you put the 1K resistor in series with the led?
Does it work now on 3.3V also?

**Reply by usr97629238** [reply]
No, currently it is wired as pictured. It doesn't work with TX/RX at 5V0, however it works just fine with TX/RX at 3V3 with either 3V3 or 5V0 VCC.

**Reply by hmeijdam** [reply]
For pulling UPDI low from 5V you probably need to remove the led or put the 1K resistor there.  Mine has a (weirdly enough shared) 1K on its RX/TX leds and does not even have the 470R in series with TX. But works at all voltages.
